app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -241,25 +245,38 @@ def build_metric_id(type, resource, metric_id)
     # by ems_ref, which in Hawkular terms is the
     # fully qualified resource path from Hawkular inventory
     def run_generic_operation(operation, ems_ref, parameters = {})
-      with_provider_connection do |connection|
         the_operation = {
           :operationName => operation,
           :resourcePath  => ems_ref.to_s,
           :parameters    => parameters
         }
+        run_operation(the_operation)
+    end
 
+    def run_specific_operation(operation_name, ems_ref, parameters = {})
+      parameters[:resourcePath] = ems_ref.to_s
+      run_operation(parameters, operation_name)
+    end
+
+    def run_operation(parameters, operation_name = nil)
+      with_provider_connection do |connection|
         actual_data = {}
-        connection.operations(true).invoke_generic_operation(the_operation) do |on|
+        callback = proc do |on|
           on.success do |data|
             _log.debug "Success on websocket-operation #{data}"
             actual_data[:data] = data
           end
           on.failure do |error|
-            actual_data[:data]  = {}
             actual_data[:error] = error
             _log.error 'error callback was called, reason: ' + error.to_s
```

Comment:
will the user be able to understand what exact operation failed from this log message?